nuxeo-features/nuxeo-automation/nuxeo-automation-scripting/src/main/java/org/nuxeo/automation/scripting/internals/AutomationScriptingServiceImpl.java

```diff
@@ -143,84 +146,46 @@ public void close() throws Exception {
         }
     }
 
-    class Factory {
-
-        final NashornScriptEngineFactory nashorn = new NashornScriptEngineFactory();
-
-        final Supplier<ScriptEngine> supplier = supplier();
-
-        Supplier<ScriptEngine> supplier() {
-
-            Log log = LogFactory.getLog(AutomationScriptingServiceImpl.class);
-            String version = Framework.getProperty("java.version");
-            // Check if jdk8
-            if (!checkJavaVersion(version, NASHORN_JAVA_VERSION)) {
-                throw new UnsupportedOperationException(NASHORN_JAVA_VERSION);
-            }
-            // Check if version < jdk8u25 -> no cache.
-            if (!checkJavaVersion(version, COMPLIANT_JAVA_VERSION_CACHE)) {
-                log.warn(NASHORN_WARN_CACHE);
-                return noCache();
-            }
-            // Check if jdk8u25 <= version < jdk8u40 -> only cache.
-            if (!checkJavaVersion(version, COMPLIANT_JAVA_VERSION_CLASS_FILTER)) {
-                if (Boolean.parseBoolean(
-                        Framework.getProperty(AUTOMATION_SCRIPTING_PRECOMPILE, DEFAULT_PRECOMPILE_STATUS))) {
-                    log.warn(NASHORN_WARN_CLASS_FILTER);
-                    return cache();
-                } else {
-                    log.warn(NASHORN_WARN_CLASS_FILTER);
-                    return noCache();
-                }
-            }
-            // Else if version >= jdk8u40 -> cache + class filter
-            try {
-                if (Boolean.parseBoolean(
-                        Framework.getProperty(AUTOMATION_SCRIPTING_PRECOMPILE, DEFAULT_PRECOMPILE_STATUS))) {
-                    return cacheAndClassFilter();
-                } else {
-                    return noCacheAndClassFilter();
-                }
-            } catch (NoClassDefFoundError cause) {
-                log.warn(NASHORN_WARN_CLASS_FILTER);
-                return cache();
-            }
+    protected ScriptEngine getScriptEngine() {
```

Comment:
Do we make the `ScriptEngine` available to any other services?  Is so, is there a public method to get a reference to it?

Reply:
There's various other places in Nuxeo that use a script engine (nuxeo-core-event, nuxeo-core-io, nuxeo-webengine-core, nuxeo-platform-renditionnuxeo-connect-client), but it's acquired through a `ScriptEngineManager`. We don't have the same class-filter restrictions in those, but maybe we should... Also the `ScriptEngineManager` is also used to acquire a Groovy engine or any other that can be configured, not just Nashorn.

These should be probably be consolidated into a single engine lookup service, but that's for another ticket.